graph/history.go

```diff
@@ -38,8 +37,7 @@ func (graph *Graph) depth(img *image.Image) (int, error) {
 
 	for parent != nil {
 		count++
-		parent, err = graph.GetParent(parent)
-		if err != nil {
+		if parent, err = graph.GetParent(parent); err != nil {
```

Comment:
Nop,,maybe it's a bug of Golang, but the compiler of Golang will not take `parent` as an `old` variable if we change these code as 
```
...
if parent, err := graph.GetParent(parent); err != nil {
}
...
```

The compiler will error out
```
graph/history.go:39: parent declared and not used
```